### PR TITLE
add NZTM2000Quad tile matrix set

### PR DESCRIPTION
**Pull request by blacha**
This imports the new NZTM2000Quad tile matrix set from LINZ.

### CODE REVIEW

**Review by geospatial-jeff — APPROVED** [review]
that decimation 😍

**Comment by vincentsarago** [comment]
@blacha thanks for the PR. Would you be able to fix and maybe add more test for your TMS ?

File: tests/test_morecantile.py [hunk 1]
```diff
@@ -10,32 +10,34 @@
 
 from .conftest import gdal_version, requires_gdal3, requires_gdal_lt_3
 
+DEFAULT_GRID_COUNT = 11
```
Comment (vincentsarago) [inline tests/test_morecantile.py:13]: ❤️

**Comment by blacha** [comment]
Im not sure about these last failing tests, they dont look to be anything i have modified?

**Comment by vincentsarago** [comment]
Yeah it seems that rasterio changed again. I'll have a look

**Comment by vincentsarago** [comment]
tests are still failing for GDAL2 but I'll go ahead and merge anyway 

🙏 thanks @blacha

**Pull request merged**